pkg/apis/storage/v1beta1/defaults.go

```diff
@@ -47,4 +47,8 @@ func SetDefaults_CSIDriver(obj *storagev1beta1.CSIDriver) {
 		obj.Spec.PodInfoOnMount = new(bool)
 		*(obj.Spec.PodInfoOnMount) = false
 	}
+	if obj.Spec.Mode == nil {
```

Comment:
I think this needs to be feature gated

Reply:
We also need to drop the alpha field in `PrepareForCreate`, `PrepareForUpdate` calls in https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/storage/csidriver/strategy.go.  Take a look at [this example](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/storage/csinode/strategy.go)

Reply:
Okay. But will this help with the roundtrip problem?

Reply:
And which part of that example are the hand-written files? They all come from https://github.com/kubernetes/kubernetes/commit/bb45b8ee34f11bf858a29c8e4176b93ed1b76928#diff-da9973f9d53ca756ac31a053eb90161c, which includes everything in a single commit.

Reply:
the entire strategy.go is a handwritten file

Reply:
To fix the roundtripping test, I think we need to modify the [fuzzer](https://github.com/kubernetes/kubernetes/blob/6c31101257bfcd47fa53702cea07fe2eedf2ad92/pkg/apis/storage/fuzzer/fuzzer.go)

Reply:
I added the feature gate check to `defaults.go` and `strategy.go`. With the feature disabled, the roundtrip test passes, but of course breaks again when enabling it by default.

> To fix the roundtripping test, I think we need to modify the fuzzer

Yes, that did the trick.

I'm going to squash the following change into the commit:
```
diff --git a/pkg/apis/storage/fuzzer/fuzzer.go b/pkg/apis/storage/fuzzer/fuzzer.go
index 2081ffe846..c205f2c4f7 100644
--- a/pkg/apis/storage/fuzzer/fuzzer.go
+++ b/pkg/apis/storage/fuzzer/fuzzer.go
@@ -46,6 +46,10 @@ var Funcs = func(codecs runtimeserializer.CodecFactory) []interface{} {
                                obj.Spec.PodInfoOnMount = new(bool)
                                *(obj.Spec.PodInfoOnMount) = false
                        }
+                       if obj.Spec.Mode == nil {
+                               obj.Spec.Mode = new(storage.CSIDriverMode)
+                               *obj.Spec.Mode = storage.PersistentDriverMode
+                       }
                },
        }
 }
diff --git a/pkg/apis/storage/v1beta1/defaults.go b/pkg/apis/storage/v1beta1/defaults.go
index 05bc6dbb3e..75acec10f3 100644
--- a/pkg/apis/storage/v1beta1/defaults.go
+++ b/pkg/apis/storage/v1beta1/defaults.go
@@ -20,6 +20,8 @@ import (
        "k8s.io/api/core/v1"
        storagev1beta1 "k8s.io/api/storage/v1beta1"
        "k8s.io/apimachinery/pkg/runtime"
+       utilfeature "k8s.io/apiserver/pkg/util/feature"
+       "k8s.io/kubernetes/pkg/features"
 )
 
 func addDefaultingFuncs(scheme *runtime.Scheme) error {
@@ -47,7 +49,7 @@ func SetDefaults_CSIDriver(obj *storagev1beta1.CSIDriver) {
                obj.Spec.PodInfoOnMount = new(bool)
                *(obj.Spec.PodInfoOnMount) = false
        }
-       if obj.Spec.Mode == nil {
+       if obj.Spec.Mode == nil && utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) {
                obj.Spec.Mode = new(storagev1beta1.CSIDriverMode)
                *(obj.Spec.Mode) = storagev1beta1.PersistentDriverMode
        }
diff --git a/pkg/registry/storage/csidriver/strategy.go b/pkg/registry/storage/csidriver/strategy.go
index 1f534b2e76..1b31a7b23a 100644
--- a/pkg/registry/storage/csidriver/strategy.go
+++ b/pkg/registry/storage/csidriver/strategy.go
@@ -22,9 +22,11 @@ import (
        "k8s.io/apimachinery/pkg/runtime"
        "k8s.io/apimachinery/pkg/util/validation/field"
        "k8s.io/apiserver/pkg/storage/names"
+       utilfeature "k8s.io/apiserver/pkg/util/feature"
        "k8s.io/kubernetes/pkg/api/legacyscheme"
        "k8s.io/kubernetes/pkg/apis/storage"
        "k8s.io/kubernetes/pkg/apis/storage/validation"
+       "k8s.io/kubernetes/pkg/features"
 )
 
 // csiDriverStrategy implements behavior for CSIDriver objects
@@ -41,8 +43,12 @@ func (csiDriverStrategy) NamespaceScoped() bool {
        return false
 }
 
-// ResetBeforeCreate clears the Status field which is not allowed to be set by end users on creation.
+// PrepareForCreate clears the Mode field if the corresponding feature is disabled.
 func (csiDriverStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
+       if !utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) {
+               csiDriver := obj.(*storage.CSIDriver)
+               csiDriver.Spec.Mode = nil
+       }
 }
 
 func (csiDriverStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
@@ -62,8 +68,12 @@ func (csiDriverStrategy) AllowCreateOnUpdate() bool {
        return false
 }
 
-// PrepareForUpdate sets the Status fields which is not allowed to be set by an end user updating a CSIDriver
+// PrepareForUpdate clears the Mode field if the corresponding feature is disabled.
 func (csiDriverStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+       if !utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) {
+               newCSIDriver := obj.(*storage.CSIDriver)
+               newCSIDriver.Spec.Mode = nil
+       }
 }
 
 func (csiDriverStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
```